include/deal.II/physics/notation.h

```diff
@@ -0,0 +1,1513 @@
+
+            for (unsigned int c=0; c<n_cols; ++c)
+              {
+                const std::pair<unsigned int,unsigned int> indices_kl = internal::indices_from_component<dim>(c,false);
+                Assert(indices_kl.first < dim, ExcInternalError());
+                Assert(indices_kl.second < dim, ExcInternalError());
+                const unsigned int &k = indices_kl.first;
+                const unsigned int &l = indices_kl.second;
+
+                out(r,c) = t[i][j][k][l];
+              }
+          }
+        return out;
+      }
+
+
+      template<int dim, typename Number>
+      FullMatrix<Number>
+      unroll_to_matrix (const SymmetricTensor<4,dim,Number> &st)
+      {
+        FullMatrix<Number> out (SymmetricTensor<2,dim,Number>::n_independent_components,
+                                SymmetricTensor<2,dim,Number>::n_independent_components);
+        const unsigned int n_rows = out.m();
+        const unsigned int n_cols = out.n();
+        for (unsigned int r=0; r<n_rows; ++r)
+          {
+            const std::pair<unsigned int,unsigned int> indices_ij = internal::indices_from_component<dim>(r,true);
+            Assert(indices_ij.first < dim, ExcInternalError());
+            Assert(indices_ij.second < dim, ExcInternalError());
+            Assert(indices_ij.second >= indices_ij.first, ExcInternalError());
+            const unsigned int &i = indices_ij.first;
+            const unsigned int &j = indices_ij.second;
+
+            for (unsigned int c=0; c<n_cols; ++c)
+              {
+                const std::pair<unsigned int,unsigned int> indices_kl = internal::indices_from_component<dim>(c,true);
+                Assert(indices_kl.first < dim, ExcInternalError());
+                Assert(indices_kl.second < dim, ExcInternalError());
+                Assert(indices_kl.second >= indices_kl.first, ExcInternalError());
+                const unsigned int &k = indices_kl.first;
+                const unsigned int &l = indices_kl.second;
+
+                const double factor = internal::matrix_component_factor<dim>(r,c,true);
+
+                out(r,c) = factor*st[i][j][k][l];
+              }
+          }
+        return out;
+      }
+
+
+      template<int dim, typename Number>
+      void
+      roll_to_tensor (Number               &s,
+                      const Vector<Number> &vec)
+      {
+        Assert(vec.size() == 1, ExcDimensionMismatch(vec.size(), 1));
+        s = vec(0);
+      }
+
+
+      template<int dim, typename Number>
+      void
+      roll_to_tensor (Tensor<0,dim,Number> &s,
+                      const Vector<Number> &vec)
+      {
+        return roll_to_tensor(s.operator Number &(), vec);
+      }
+
+
+      template<int dim, typename Number>
+      void
+      roll_to_tensor (Tensor<1,dim,Number> &v,
+                      const Vector<Number> &vec)
+      {
+        Assert(vec.size() == v.n_independent_components, ExcDimensionMismatch(vec.size(), v.n_independent_components));
```

Comment:
could you please move `Exc...` on the new line? Makes reading a bit easier. I think that's how we do it elsewhere if variable names are longish.